### ISSUE TITLE
General wiki formatting/editing

### ISSUE THREAD

**Issue by jennparker** [silexlabs/Silex#585]
Go through the documentation wiki to: 
- [ ] make style more consistent across all pages
- [ ] update links
- [ ] smooth out language for ease of reading

**Comment by lexoyo** [comment]
@jennparker thank you for your help on the wiki!!!

i am working on a new feature in silex: a new "component system". it means there are several new components (slideshow, share bar, animated container, contact form and map) and there is a new panel to edit their properties

you can see it in action here: http://preprod.silex.me/ (this is the test server)
go to the "+" menu, add a component (e.g. map) and double click on it, you will see its params on the right in place of the properties panel

i am willing to write some documentation for each component, do you have any advice?

note that this there is something about components in this page about [silex elements](https://github.com/silexlabs/Silex/wiki/Silex-Elements)

**Comment by jennparker** [comment]
@lexoyo, looks cool! Thanks for writing some documentation! To start, we should: 
1) Update the first screenshot on the [elements page](https://github.com/silexlabs/Silex/wiki/Silex-Elements), so we're showing the new elements menu with the additional items 
2) For each element:
     a. add a short description of when to use this element
     b. show a screenshot showing an example of the element on the canvas
     c. show a screenshot showing the menu that is displayed when that element on the canvas is double clicked. 
     d. add a short description of the options displayed in the element menu (from step c.)

Not related to the new items, but since we're talking about this page :-) ....
3) I am a little confused about [components](https://github.com/silexlabs/Silex/wiki/Silex-Elements#Components). Is it fair to say that components are groups of elements? Some clarification on what components are, and maybe some screenshots (if applicable), would be awesome!
4) Should this be [Prototype](https://github.com/silexlabs/Prodotype)? I wasn't sure if we are intentionally spelling it differently or if it is a typo? 

That would be an excellent start!

**Comment by lexoyo** [comment]
> For each element:
> a. add a short description of when to use this element

very good idea! (I was thinking only of describing the element)

> 3) I am a little confused about components. Is it fair to say that components are groups of elements? Some clarification on what components are, and maybe some screenshots (if applicable), would be awesome!
> 4) Should this be Prototype? I wasn't sure if we are intentionally spelling it differently or if it is a typo?

my bad, this is maybe not useful to have elements and components
in my mind elements are the base bricks, and also it should (will hopefully one day) be just an html element (container = div, htmlbox = iframe, text box = p, image = img)

whereas components are
* more complex and composed of elements
* generated with Prodotype (a pun with production and prototype)
* they have parameters (the new parameters panel)

maybe we should avoid this distinction and talk only about elements?

**Comment by lexoyo** [comment]
here is a new screenshot, is it ok @jennparker ?
![screenshot from 2017-04-19 19-51-25](https://cloud.githubusercontent.com/assets/715377/25194146/c845dc32-2539-11e7-9415-1e606d40c71d.png)

**Comment by David1platt** [comment]
Hi I'm in college studying profressional/tech writing, and I just started talking with lexoyo about working with the team. I already went through last night and made updates to the Dashboard wiki.

**Comment by jennparker** [comment]
@lexoyo, the screenshot looks great!! Thanks for the clarification on elements vs components. This is more clear now. I think if we massage this page a bit we'll be able to really make a clear distinction.  

Hi @David1platt , that's awesome! I just recently joined the group as well. This project is like heaven for tech writers because there is so much great stuff waiting to be written. I'm a former mobile dev and current freelance web dev looking to transition into a full time tech writing role here in Seattle, so I am helping out as I am able in between other projects.

**Comment by lexoyo** [comment]
Hello,
Where are we in the road map? Maybe we should make a help wanted issue per remaining thing?

**Comment by David1platt** [comment]
I just finished the text for the Getting Started page. I tried to invite an acquaintance to get involved, but I don't know what happened. I'm also kinda waiting for the new version before I get to far into further documentation.

**Comment by lexoyo** [comment]
Hi @David1platt 
there is always a new version coming and it is available on preprod.silex.me if you want to have a look

**Comment by lexoyo** [comment]
I have added pages for components:

* https://github.com/silexlabs/Silex/wiki/Live-map-component
* https://github.com/silexlabs/Silex/wiki/Skill-bar-component
* https://github.com/silexlabs/Silex/wiki/Slideshow-component
* https://github.com/silexlabs/Silex/wiki/Share-bar-component
* https://github.com/silexlabs/Silex/wiki/Contact-form-component

Most of the documentation is missing (params and technical details) and the screenshot are weird, but it's a start, a call to action for new contributors

Oh and there are links to these pages in silex, see the parameters panel of each component

**Comment by David1platt** [comment]
Is there a folder in the project where documentation people can submit their own screenshots and animated GIFs?

**Comment by lexoyo** [comment]
The way I do it is upload the image or gif in a new issue which I never submit. I copy the link to the image on github. Do you see what I mean?

**Comment by JbIPS** [comment]
Github has a storage of its own. If you drop an image into a text field it will convert it into a link to the uploaded file on Github server.

**Comment by jennparker** [comment]
As part of the restructuring, I'd like to propose that we get rid of all of the individual UI pages and instead add all of the basic UI descriptions to one page, the [Editor UI page](https://github.com/silexlabs/Silex/wiki/Editor-UI). Tutorials or guides would not go here. We would give a tutorial its own page and then just link to it in the appropriate location on the editor UI page. The page will be long, but by using anchors, it won't be challenging to find things. It seems to be standard practice in a lot of the well-organized GitHub wikis I've reviewed.

I started doing this work the week before last, so we could see what it looks like, but ran out of time to get feedback before going on vacation. The current work is on the [Editor UI page](https://github.com/silexlabs/Silex/wiki/Editor-UI). It doesn't include @lexoyo's latest component work, but I added all of the other UI-related info from the individual UI pages to this page, organized it, and added anchors. I'd like to complete this task by:

* Adding all UI descriptions to the Editor UI page  
	* As mentioned, this is already done except for pages added/edited in the last week  
* Deleting all of the individual UI pages ([Dashboard](https://github.com/silexlabs/Silex/wiki/Dashboard), [Context Menu](https://github.com/silexlabs/Silex/wiki/Context-Menu), [Stage](https://github.com/silexlabs/Silex/wiki/Stage), etc.) 
* Moving remaining how-tos out of the Editor UI page and into individual pages, which will be listed under [Tutorials and Guides](https://github.com/silexlabs/Silex/wiki/Tutorials-and-Guides) and linked in the appropriate places in the UI Editor. 

I should have time to do all of this work tomorrow but wanted to check in and get input first, since it will involve significant restructuring.

**Comment by David1platt** [comment]
Yup that totally works! Otherwise an advertisement would be useful. We need more help.

**Comment by lexoyo** [comment]
1 long page for UI and 1 for a tutorial sounds great! I can move my "components" pages into this page if you want?

> I should have time to do all of this work tomorrow but wanted to check in and get input first, since it will involve significant restructuring.

I completely agree with this plan

> Yup that totally works! Otherwise an advertisement would be useful. We need more help.

Maybe a blog post about contributing to the docs on silexlabs.org (which has 2000 unique users a month) ?

**Comment by jennparker** [comment]
Cool! @lexoyo, sounds great, please do! I'll go ahead and make sure we have everything off of the other individual UI pages and then will delete them. After that, I can write up a quick blog post asking for help unless anyone else gets to it first.

**Comment by JbIPS** [comment]
> Maybe a blog post about contributing to the docs on silexlabs.org (which has 2000 unique users a month) ? 

What about an announcement when publishing the website? At least pointing to the new doc, telling people we're creating a new one. I hope the most curious will take a look and at least give feedback

**Comment by lexoyo** [comment]
> Cool! @lexoyo, sounds great, please do! I'll go ahead and make sure we have everything off of the other individual UI pages and then will delete them. After that, I can write up a quick blog post asking for help unless anyone else gets to it first.

I'm not good at writing this kind of article, but maybe @David1platt with @JbIPS ?

**Comment by David1platt** [comment]
Two larger documents can be beneficial in minimizing duplicated work. Is there a way to create links to different sections of an individual document? 

I was thinking once the wiki is solid, it could be converted into an extension of the silexlabs webpage, that would allow us to structure the information better for ease of navigation. What are people's thoughts?

**Comment by singchan** [comment]
@jennparker I found several links hard coded in Silex to these URLs

I will replace it with `https://github.com/silexlabs/Silex/wiki/Editor-UI#dashboard`:

* `https://github.com/silexlabs/Silex/wiki/Dashboard`

will someone move it somewhere to the wiki? If you do it tell me, I will update the link in silex

* `https://github.com/silexlabs/Silex/issues/184`
* `https://github.com/silexlabs/Silex/wiki/Contact-form-component` and all the components
* `https://github.com/silexlabs/Silex/issues/184#properties-css-panel`
* `https://github.com/silexlabs/Silex/issues/238`
* `https://github.com/silexlabs/Silex/issues/199`
* `https://github.com/silexlabs/Silex/issues/184`
* `https://github.com/silexlabs/Silex/issues/200`
* `https://github.com/silexlabs/Silex/issues/282`

**Comment by jennparker** [comment]
@singchan thanks for this! Moving these issues to the wiki is still in progress. I'll go ahead and open an issue so we can easily track all of the issues that need to be moved to the wiki.
